crates/bevy_feathers/src/controls/slider.rs

```diff
@@ -0,0 +1,214 @@
+    schedule::IntoScheduleConfigs,
+    spawn::SpawnRelated,
+    system::{In, Query, Res, SystemId},
+};
+use bevy_input_focus::tab_navigation::TabIndex;
+use bevy_picking::PickingSystems;
+use bevy_ui::{
+    widget::Text, AlignItems, BackgroundGradient, ColorStop, Display, FlexDirection, Gradient,
+    InteractionDisabled, JustifyContent, LinearGradient, Node, UiRect, Val,
+};
+use bevy_winit::cursor::CursorIcon;
+
+use crate::{
+    font_styles::InheritableFont,
+    handle_or_path::HandleOrPath,
+    theme::{
+        self,
+        corners::RoundedCorners,
+        fonts, size,
+        tokens::{SLIDER_BAR, SLIDER_BAR_DISABLED, SLIDER_BG},
+        ThemeFontColor, UiTheme, UseTheme,
+    },
+};
+
+/// Slider template properties, passed to [`slider`] function.
+// TODO: These should all be replaced with reactive signals.
+#[derive(Clone)]
+pub struct SliderProps {
+    /// Slider current value
+    pub value: f32,
+    /// Slider minimum value
+    pub min: f32,
+    /// Slider maximum value
+    pub max: f32,
+    /// Number of decimal digits of precision to display.
+    pub precision: usize,
+    /// On-change handler
+    pub on_change: Option<SystemId<In<f32>>>,
+}
+
+impl Default for SliderProps {
+    fn default() -> Self {
+        Self {
+            value: 0.0,
+            min: 0.0,
+            max: 1.0,
+            precision: 1,
+            on_change: None,
+        }
+    }
+}
+
+#[derive(Component, Default, Clone)]
+#[require(CoreSlider)]
+struct SliderStyle;
+
+/// Marker for the text
+#[derive(Component, Default, Clone)]
+struct SliderValueText;
+
+/// Spawn a new slider widget.
+///
+/// # Arguments
+///
+/// * `props` - construction properties for the slider.
+/// * `overrides` - a bundle of components that are merged in with the normal slider components.
+pub fn slider<B: Bundle>(props: SliderProps, overrides: B) -> impl Bundle {
+    (
+        Node {
+            height: size::ROW_HEIGHT,
+            justify_content: JustifyContent::Center,
+            align_items: AlignItems::Center,
+            padding: UiRect::axes(Val::Px(8.0), Val::Px(0.)),
+            flex_grow: 1.0,
+            ..Default::default()
+        },
+        CoreSlider {
+            on_change: props.on_change,
+            track_click: TrackClick::Drag,
+        },
+        SliderStyle,
+        SliderValue(props.value),
+        SliderRange::new(props.min, props.max),
+        CursorIcon::System(bevy_window::SystemCursorIcon::EwResize),
+        TabIndex(0),
+        RoundedCorners::All.to_border_radius(6.0),
+        // Use a gradient to draw the moving bar
+        BackgroundGradient(vec![Gradient::Linear(LinearGradient {
+            angle: PI * 0.5,
+            stops: vec![
+                ColorStop::new(Color::NONE, Val::Percent(0.)),
+                ColorStop::new(Color::NONE, Val::Percent(50.)),
+                ColorStop::new(Color::NONE, Val::Percent(50.)),
+                ColorStop::new(Color::NONE, Val::Percent(100.)),
```

Comment:
The the zero and one hundred percent stops can be elided here. When there is space before the first stop or after the last stop it fills with the respective stop's color.

Reply:
I like that this is more explicit.

Reply:
I guess so, but it's also a little more complicated to update the colors as there are four values to change instead of two.